src/sections/SideBar/components/Footer/SideBarFooter.js

```diff
@@ -0,0 +1,92 @@
+import React from "react";
+import { makeStyles, Link, Typography } from "@material-ui/core";
+import { FacebookIcon, InstagramIcon, LinkedInIcon, GitHubIcon } from '@material-ui/icons';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    display: "flex",
+    justifyContent: "center",
+    flexDirection: "column",
+    width: "16%",
+    minWidth: 200,
+    maxWidth: 240,
+    backgroundColor: "rgb(61, 61, 61)",
+    border: "1px solid black",
+  },
+  container: {
+    display: "flex",
+    justifyContent: "center"
+  },
+  text: {
+    color: "gray",
+    textAlign: "center",
+    paddingBottom: 15
+  },
+  icon: {
+    color: "gray",
+    margin: 5,
+    marginBottom: 15
+  },
+  icon2: {
+    color: "white",
+    margin: 5,
+    marginBottom: 15
+  }
+}));
+
+const Footer = () => {
+  const classes = useStyles();
+  const [selectedIndex, setSelectedIndex] = React.useState(0);
+
+  const handleMouseOver = (event, index) => {
+    setSelectedIndex(index);
+  };
+
+  const selectIconColor = (index) => {
+    return (selectedIndex === index) ? classes.icon2 : classes.icon;
+  };
+
+  const arrValues = [["https://www.facebook.com/", <FacebookIcon 
+  className={selectIconColor(1)} 
+  onMouseOver={(event) => handleMouseOver(event, 1)} 
+  onMouseOut={(event) => handleMouseOver(event, 0)} 
+  />], 
+  ["https://www.instagram.com/", <InstagramIcon 
```

Comment:
Seems like a component, walks like a component and quacks like one then let's make this a Icon component that accepts an argument string to specify which icon is going to render and also receives both events. By doing this you should be able to avoid setting the component inside the array and just rendering that generic icon component